website/blog/2025-04-20-vs-code-with-podman-desktop.md

```diff
@@ -0,0 +1,100 @@
+
+![enabling docker compatibility in the settings](img/vs-code-podman/docker-compatibility.png)
+
+### Option 2: Pod Manager
+
+To install the extension:
+
+1. Open VS Code.
+2. Click on the Extensions icon in the Activity Bar (or press `Ctrl+Shift+X` or `Cmd+Shift+X`).
+3. Search for "Pod Manager" and install the extension by dreamcatcher45.
+
+## Using the VS Code Extensions
+
+Now that we have the extensions installed and configured, let's see how to use them.
+
+### Working with Containerfiles and Dockerfiles
+
+Both extensions provide syntax highlighting, code completion, and linting for Containerfiles and Dockerfiles. Open a Containerfile in VS Code, and you'll immediately benefit from these features.
+
+You can also build images directly from VS Code:
+
+1. Right-click on the Containerfile in the Explorer view.
+2. Select "Build Image".
+3. VS Code will prompt you for an image name and tag.
+4. The extension will then build the image using Podman.
+
+![building a Containerfile in vs code using the microsoft extension](img/vs-code-podman/build-image.png)
+
+### Managing Containers
+
+The extensions also allow you to manage containers directly from VS Code. You can start, stop, restart, and remove containers, as well as view their logs and inspect their configuration.
+
+To view the container logs:
+
+1. Click on the Container icon in the Activity Bar.
+2. You'll see a list of your containers, images, and networks.
+3. Right-click on a container to perform actions.
+
+![using Container Tools extension to view the logs of the container](img/vs-code-podman/view-logs.png)
+
+Similarly using the Pod manager extension we can visually inspect containers, images, and volumes.
+
+1. Click on the Pod manager icon in the Activity Bar.
+2. You'll see a list of your containers, images, and networks.
+
+![using podmanager to view all the running containers, images and volumes](img/vs-code-podman/podmanager-details.png)
+
+and manage the container lifecycle.
+
+![using podmanager to manage the lifecycle of the container](img/vs-code-podman/interact-with-container.png)
+
+and of course troubleshoot issues with a visual interface.
+
+![using podmanager to enter the container](img/vs-code-podman/podmanager-details.png)
+
+## Conclusion
+
+By combining the power of VS Code, Podman, and Podman Desktop, you can create a streamlined and efficient container development workflow. The VS Code extensions provide excellent integration with Podman, allowing you to manage containers, build images, and work with Containerfiles directly from your code editor. Podman Desktop complements this with a visual interface for managing your container environment. Embrace these tools and elevate your container development experience!
```

Comment:
I think it would be nice to expand on this particular aspect:
 > Podman Desktop complements this with a visual interface for managing your container environment

There could be a dedicated section showing how to work with the two tools side by side. What are the advantages of one compare to the other?

Reply:
Honestly they are nearly identical in feature set. I added a little line for it though.

Reply:
This paragraphe is about the complementary of VSCode + Podman + Podman Desktop. So my point is not about the two extensions. 

It's about how Podman Desktop and one of these extensions are making a developer comfortably well tooled, how one should be using the two tools so that they get an "elevated development experience". This is not covered in the blog post and right now it conveys the message that, if you are using VSCode with one of this extension, you don't really see the point of using Podman Desktop anymore.

Reply:
Ok, I added a line mentioning that Podman Desktop will continue where your development has left off and take you to production.